### ISSUE TITLE
when trying to build sample-app, error about imagerepository occurs

### ISSUE THREAD

**Issue by thoraxe** [openshift/origin#1619]
```
Apr 06 17:20:32 ose3-master.example.com openshift[5886]: I0406 17:20:32.365619    5886 controller.go:42] Handling build ruby-sample-build-1
Apr 06 17:20:32 ose3-master.example.com openshift[5886]: E0406 17:20:32.366458    5886 factory.go:30] Build failed with error integrated/ruby-sample-build-1: &errors.errorString{s:"the referenced output image repository integrated/origin-ruby-sample does not exist"}
```

This just repeats ad-infinitum.

I have created a template like so:

```
osc create -f https://raw.githubusercontent.com/openshift/origin/master/examples/sample-app/application-template-stibuild.json -n openshift
```

And then I am trying to create an instance of this template. When starting the build, the error above occurs.

Note, this may also somehow be related to #1615

**Comment by soltysh** [comment]
This creates *just* the template object itself and not the objects from that template. I just did that:
```
$ osc create -f application-template-stibuild.json -n openshift
templates/ruby-helloworld-sample
```
You should have a look at [sample-app](https://github.com/openshift/origin/tree/master/examples/sample-app) step 15, which first of all processes the template (`osc process` part) and only after creates (`osc create`) the objects, see below output:
```
$ osc process -f application-template-stibuild.json | osc create -f -
services/frontend
routes/route-edge
imageStreams/origin-ruby-sample
imageStreams/ruby-20-centos7
buildConfigs/ruby-sample-build
deploymentConfigs/frontend
services/database
deploymentConfigs/database

```

**Comment by thoraxe** [comment]
Clicking the "create" button in the web console should be the same thing as "osc process -f template | osc create -f -".

Otherwise, what is the purpose of "osc create -f template"?

CLI and the WebUI "experience" are different.

But it appears that something more sinister is going on on the back-end when using the WebUI to create an instance of a template.

I think you are talking about something different than the bug references.

**Comment by soltysh** [comment]
On Tue, Apr 7, 2015 at 3:57 PM Erik M Jacobs <notifications@github.com>
wrote:

> Clicking the "create" button in the web console should be the same thing
> as "osc process -f template | osc create -f -".
>
> Otherwise, what is the purpose of "osc create -f template"?
>
This creates the template object, the same way you create other objects. To
create objects from that template you need to process the template which in
result gives you a Config, which under the hood is a list of objects to be
created.

> CLI and the WebUI "experience" are different.
>


> But it appears that something more sinister is going on on the back-end
> when using the WebUI to create an instance of a template.
>

I think you are talking about something different than the bug references.

**Comment by thoraxe** [comment]
The issue this bug describes is that, after creating a template, when it is instantiated from the web UI and then the build is started, the error in the first comment is found -- the build will never complete.

**Comment by thoraxe** [comment]
Still happening with latest puddle:

```
osc v0.4.2.4
kubernetes v0.13.1-dev-641-gf057a25
```

**Comment by smarterclayton** [comment]
The error you're seeing may be due to a change in how image streams were created.  @ncdc

**Comment by ncdc** [comment]
Working on a fix now.

**Comment by ncdc** [comment]
I believe this is also fixed by #1638

**Comment by danmcp** [comment]
Any remaining issues here?
